public/app/features/plugins/ds_edit_ctrl.ts

```diff
@@ -120,6 +122,27 @@ export class DataSourceEditCtrl {
     });
   }
 
+  onAfterSave() {
+    return this.datasourceSrv.get(this.current.name).then(datasource => {
+      if (!datasource.onAfterSave) {
+        return;
+      }
+
+      return datasource
+        .onAfterSave(this.current)
+        .then(result => {
+          if (result.changed === true) {
+            return this.backendSrv.put('/api/datasources/' + this.current.id, this.current).then(result => {
```

Comment:
I would just say to place the `.then` statement on the next line to be consistent.